src/base_authentication_app_skeleton/src/forms/password_reset_request_form.cr

```diff
@@ -1,9 +1,21 @@
 class PasswordResetRequestForm < Avram::VirtualForm
-  include Authentic::FormHelpers
-  include FindAuthenticatable
+  # You can modify this in src/forms/mixins/user_from_email.cr
+  include UserFromEmail
 
   virtual email : String
 
+  # Run validations and yield the form and the user if valid
+  def submit
+    user = user_from_email
+    validate(user)
+
+    if valid?
+      yield self, user
+    else
+      yield self, nil
+    end
+  end
```

Comment:
Moved here so it is easier to see what is going on, and how the form works. Thoughts?

Reply:
Yup, I think having this abstraction live here is better than in a library. This feels like the tradeoffs that [Oath] makes where most of the user-level logic lives in the app itself doesn't force you into any particular pattern. I like that kind of tradeoff.

[Oath]: https://github.com/halogenandtoast/oath

Reply:
Is this clearer?

```crystal
  def submit
    user = user_from_email
    validate_credentials(user)
    # set user to `nil` instead of using an if with 2 yields
    user = nil unless valid?
    yield self, user
  end
```

Reply:
@edwardloveall I just added another comment here. Thoughts?

Reply:
I think the first one is clearer, although this is a bit prettier. This makes it slightly more difficult to keep track of what state `user` is in.

Reply:
I agree. I want to find a simpler way (I had an idea for something in 
https://github.com/luckyframework/avram/issues/69) but I think you're right that the conditionals make it clearer 👍